File: leetcode/0387-first-unique-character-in-a-string/solution.py
class Solution(object):
    def firstUniqChar(self, s):
        lettercount={}
        l=[]
        for i in s:
            if i not in lettercount:
                lettercount[i]=1
            else:
                lettercount[i]+=1
        for j in lettercount:
            if lettercount[j]==1:
                l.append(s.index(j))
        if l==[]:
            return -1
        else:
            return min(l)


        """
        :type s: str
        :rtype: int
        """
        

File: leetcode/0387-first-unique-character-in-a-string/solution.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> count;
        for (char ch:s){
            count[ch]+=1;
        }
        vector<int> ones;
        for (const auto &p:count){
            if (p.second==1){
                ones.push_back(s.find(p.first));
            }
        }
        if (ones.size()==0){
            return -1;
        }
    auto it= min_element(ones.begin(),ones.end());
    int ans=*it;
    return ans;
    }
};
